### PR TITLE
second moment of area of 2d polygon

### PR DESCRIPTION
**Pull request by rushyam**
implemented the function for the second moment and product moment of area of a polygon. which can be used in sympy continuum mechanics for normal stress calculation in beam.
@moorepants  @ylemkimon  @jksuom please review it.

Completes PR #13665

### CODE REVIEW

File: sympy/geometry/polygon.py
Comment (gxyd) [inline sympy/geometry/polygon.py]: A little improved wording here, how about?
```
if None, it will be calculated about the centroid of the polygon.
```

File: sympy/geometry/polygon.py
Comment (gxyd) [inline sympy/geometry/polygon.py]:
```
p: Point, size-2 tuple/list of sympifiable objects, or None (default=None)
```

File: sympy/geometry/polygon.py
Comment (gxyd) [inline sympy/geometry/polygon.py]: Put this under a new heading `Returns`.

File: sympy/geometry/polygon.py
Comment (gxyd) [inline sympy/geometry/polygon.py]: Keep the comment regarding `if p is None ... yada yada` here under `Notes` heading.

File: sympy/geometry/polygon.py
Comment (gxyd) [inline sympy/geometry/polygon.py]: Why explicit mention of "two dimensional", a polygon is always 2-D?

**Review by gxyd — CHANGES_REQUESTED** [review]
Thanks for the PR. I've made a few necessary documentation necessary comments.

I'm not sure about the formula correctness.

Also please add a "Reference heading" pointing to the appropriate references.

File: sympy/geometry/polygon.py
Comment (gxyd) [inline sympy/geometry/polygon.py]: instead use `if p is None:`

**Comment by rushyam** [comment]
https://github.com/sympy/sympy/pull/13665 this pr was accidentally closed So i created a new pr

File: sympy/geometry/polygon.py
Comment (rushyam) [reply sympy/geometry/polygon.py]: yes it looks good

**Comment by gxyd** [comment]
You should use the PR template appropriately, may be you can mention that PR in the PR description. That will be helpful.

File: sympy/geometry/polygon.py
Comment (rushyam) [reply sympy/geometry/polygon.py]: it was suggested by @moorepants and the definition of Polygon I know is a plane bounded by at least three straight sides and angles. so it is possible that we can have a polygon in 3d.

File: sympy/geometry/polygon.py
Comment (gxyd) [inline sympy/geometry/polygon.py]: Having default as `p='centroid'` seems more understandable, instead of `None`. And hence update the docstring as well.

File: sympy/geometry/polygon.py
Comment (gxyd) [inline sympy/geometry/polygon.py]: moments or moment?
And why 'products' of area?

File: sympy/geometry/tests/test_polygon.py
Comment (gxyd) [inline sympy/geometry/tests/test_polygon.py]: add a space between '#' and 'rectangle.

File: sympy/geometry/tests/test_polygon.py
Comment (gxyd) [inline sympy/geometry/tests/test_polygon.py]: add a space here between '#' and 'triangle'

File: sympy/geometry/tests/test_polygon.py
Comment (gxyd) [inline sympy/geometry/tests/test_polygon.py]: space between `p` and `=` and `=` and `(0, 0)`.

File: sympy/geometry/tests/test_polygon.py
Comment (gxyd) [inline sympy/geometry/tests/test_polygon.py]: space here

**Review by gxyd — COMMENTED** [review]
Please add a "Reference" section as well.

File: sympy/geometry/polygon.py
Comment (gxyd) [inline sympy/geometry/polygon.py]: 'centroid' instead of `None`.

File: sympy/geometry/polygon.py
Comment (gxyd) [inline sympy/geometry/polygon.py]: the information present in the `Notes` section should be shifted here instead. i.e. when `p='centroid'` how it is calculated.

File: sympy/geometry/polygon.py
Comment (gxyd) [inline sympy/geometry/polygon.py]: I think we can remove this `Notes` section and instead move it where `p` is described.

**Review by gxyd — CHANGES_REQUESTED** [review]
Otherwise LGTM

File: sympy/geometry/polygon.py
Comment (gxyd) [inline sympy/geometry/polygon.py]: 'centroid' (missed inverted commas) i.e with string centroid.

File: sympy/geometry/polygon.py
Comment (gxyd) [inline sympy/geometry/polygon.py]: 1. If (i.e. capital I)
2. ``p='centroid'``

making it: If \`\`p='centroid'\`\` it will be calculated ...

**Review by gxyd — COMMENTED** [review]
After this PR is merged. Please add an entry to the change log in https://github.com/sympy/sympy/wiki/Release-Notes-for-1.2#new-features. Like the other entries there, please reference this pull request with PR number and credit yourself (and other contributors if applicable).

**Review by gxyd — APPROVED** [review]

**Comment by gxyd** [comment]
Looks good. Will merge in 24 hours if no one objects.

**Comment by rushyam** [comment]
thank you for guiding me @gxyd

**Review by sanskar107 — CHANGES_REQUESTED** [review]
Looks nice

**Comment by gxyd** [comment]
Thanks @ylemkimon . Thanks @rushyam for working on this, I made a slight modification in release notes entry by adding 'geometry' as heading instead of issue/PR heading.

**Pull request merged**
